aws/sdk/examples/autoscaling/src/bin/create-autoscaling-group.rs

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use autoscaling::{Client, Config, Error, Region, PKG_VERSION};
+use aws_types::region::ProvideRegion;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+struct Opt {
+    /// The name of the AutoScaling group.
+    #[structopt(short, long)]
+    autoscaling_name: String,
+
+    /// The ID of the EC2 instance to add to the AutoScaling group.
+    #[structopt(short, long)]
+    instance_id: String,
+
+    /// The AWS Region.
+    #[structopt(short, long)]
+    region: Option<String>,
+
+    /// Whether to display additional information.
+    #[structopt(short, long)]
+    verbose: bool,
+}
+
+/// Creates an AutoScaling group in the Region.
+/// # Arguments
+///
+/// * `-a AUTOSCALING-NAME` - The name of the AutoScaling group.
+/// * `-i INSTANCE-ID` - The ID of the Ec2 instance to add to the AutoScaling group.
+/// * `[-d DEFAULT-REGION]` - The Region in which the client is created.
+///    If not supplied, uses the value of the **AWS_REGION** environment variable.
+///    If the environment variable is not set, defaults to **us-west-2**.
+/// * `[-v]` - Whether to display additional information.
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    tracing_subscriber::fmt::init();
+
+    let Opt {
+        autoscaling_name,
+        instance_id,
+        region,
+        verbose,
+    } = Opt::from_args();
+
+    let region = region
```

Comment:
Take a look at the [cognitoidentity](https://github.com/awslabs/smithy-rs/blob/main/aws/sdk/examples/cognitoidentity/src/bin/list-identity-pools.rs#L38-L48) example for how we should be acquiring the region now. We now have a region ChainProvider that simplifies it.